From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Wed, 21 Jun 2023 11:01:56 -0400
Subject: Switch to webkit 4.1

webkit2gtk 4.1 (and Debian's librest) uses libsoup3

Also, disable the keyboard shortcut for qrencode since
that feature hasn't been ported to libsoup3 yet

https://github.com/pdfpc/pdfpc/issues/664
---
 README.rst         | 4 ++--
 rc/pdfpcrc         | 2 +-
 src/CMakeLists.txt | 6 +++---
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/README.rst b/README.rst
index d4ae468..958b9c1 100644
--- a/README.rst
+++ b/README.rst
@@ -105,11 +105,11 @@ In order to compile and run pdfpc, the following requirements need to be met:
 - libsoup
 - libqrencode
 
-E.g., on Ubuntu 18.04 onward, you can install these dependencies with::
+E.g., on Ubuntu 22.04 onward, you can install these dependencies with::
 
     sudo apt-get install cmake valac libgee-0.8-dev libpoppler-glib-dev
     libgtk-3-dev libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev
-    libjson-glib-dev libmarkdown2-dev libwebkit2gtk-4.0-dev libsoup2.4-dev
+    libjson-glib-dev libmarkdown2-dev libwebkit2gtk-4.1-dev
     libqrencode-dev gstreamer1.0-gtk3
 
 (the last one is a run-time dependence). You should also consider installing all
diff --git a/rc/pdfpcrc b/rc/pdfpcrc
index c38be5f..8779ad6 100644
--- a/rc/pdfpcrc
+++ b/rc/pdfpcrc
@@ -118,7 +118,7 @@ bind d                  toggleDrawings
 bind S+x                executeScript
 
 # Show QR code
-bind Q                  showQRcode
+# bind Q                  showQRcode
 
 # Short help
 bind question           showHelp
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 028e61f..b273de6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -30,8 +30,8 @@ if (MOVIES)
 endif ()
 
 if (MDVIEW)
-    pkg_check_modules(WEBKIT REQUIRED webkit2gtk-4.0)
-    set(MDVIEW_PACKAGES webkit2gtk-4.0)
+    pkg_check_modules(WEBKIT REQUIRED webkit2gtk-4.1)
+    set(MDVIEW_PACKAGES webkit2gtk-4.1)
     set(EXTRA_VALA_OPTIONS ${EXTRA_VALA_OPTIONS} -D MDVIEW)
 endif ()
 
@@ -136,7 +136,7 @@ endif()
 vala_precompile(VALA_C
     ${VALA_SRC}
 PACKAGES
-    webkit2gtk-4.0
+    webkit2gtk-4.1
     json-glib-1.0
     gio-2.0
     gee-0.8
